lib/spack/spack/cmd/release_jobs.py

```diff
@@ -0,0 +1,406 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import re
+
+import subprocess
+from jsonschema import validate
+from six import iteritems
+
+import llnl.util.tty as tty
+
+from spack.architecture import sys_type
+from spack.dependency import all_deptypes
+from spack.spec import Spec
+from spack.paths import spack_root
+from spack.error import SpackError
+from spack.schema.os_container_mapping import schema
+from spack.util.spec_set import CombinatorialSpecSet
+import spack.util.spack_yaml as syaml
+
+description = "generate release build set as .gitlab-ci.yml"
+section = "build"
+level = "long"
+
+
+CMD_ERROR_OUTPUT_REGEX = re.compile(
+    r'==>\s+Error:\s+(.+)<BEGIN_SPACK_COMMAND_OUTPUT>',
+    re.MULTILINE | re.DOTALL)
+IGNORE_SURROUND_REGEX = re.compile(
+    r'<BEGIN_SPACK_COMMAND_OUTPUT>(.+)<END_SPACK_COMMAND_OUTPUT>',
+    re.MULTILINE | re.DOTALL)
+DEP_LINE_REGEX = re.compile(r'^([^\s]+) -> (.+)$', re.MULTILINE)
+SPEC_LINE_REGEX = re.compile(r'^label: ([^,]+), spec: (.+)$', re.MULTILINE)
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-s', '--spec-set', default=None,
+        help="path to release spec-set yaml file")
+
+    subparser.add_argument(
+        '-m', '--mirror-url', default='http://172.17.0.1:8081/',
+        help="url of binary mirror where builds should be pushed")
+
+    subparser.add_argument(
+        '-o', '--output-file', default=".gitlab-ci.yml",
+        help="path to output file to write")
+
+    subparser.add_argument(
+        '-t', '--shared-runner-tag', default=None,
+        help="tag to add to jobs for shared runner selection")
+
+    subparser.add_argument(
+        '-k', '--signing-key', default=None,
+        help="hash of gpg key to use for package signing")
+
+    subparser.add_argument(
+        '-c', '--cdash-url', default='https://cdash.spack.io',
+        help="Base url of CDash instance jobs should communicate with")
+
+    subparser.add_argument(
+        '-p', '--print-summary', action='store_true', default=False,
+        help="Print summary of staged jobs to standard output")
+
+    subparser.add_argument(
+        '--spec-deps', default=None,
+        help="The spec for which you want container-generated deps")
+
+    subparser.add_argument(
+        '--this-machine-only', action='store_true', default=False,
```

Comment:
I think a clearer option name would be `--concretize-locally` or `--resolve-deps-locally`, since this just determines the dependencies locally (the description of the option is informative but the option could be named more-suggestively)